### PR TITLE
Identify AHSAP properties and update reporting views

### PR DESCRIPTION
**Pull request by wrridgeway**
This PR adds one new column to `reportingvw_pin_township_class`, an AHSAP indicator based on values from a number of different iasWorld fields:

> land: infl1 IN (30, 31, 32) OR SUBSTR(user7, 1, 3) = 'SAP'
> infl1 is any of the codes 30, 31, 32, or user7 (incentive number) starts with 'SAP'
> 
> oby: SUBSTR(user16, 1, 2) = 'AI' OR SUBSTR(user3, 1, 3) = 'SAP'
> user16 (alt CDU) starts with 'AI' or user3 (incentive number) starts with 'SAP'
> 
> comdat: SUBSTR(user16, 1, 2) = 'AI' OR SUBSTR(user4, 1, 3) = 'SAP' OR user13 IN (49, 50, 51)
> user16 (alt CDU) starts with 'AI' or user4 (incentive number) starts with 'SAP' or user13 (subclass 2) is any of the codes 49, 50, 51
> 
> aprval: ecf IS NOT NULL
> ecf has any value in it
> 
> dweldat: SUBSTR(user16, 1, 2) = 'AI'
> user16 (alt CDU) starts with 'AI'

### CODE REVIEW

**Comment by wrridgeway** [comment]
This query should return nothing if out additions didn't change the view:

```
WITH old AS (

    SELECT
        *
    FROM reporting.vw_pin_township_class
),

new AS (

    SELECT
        pin,
        year,
        triad_name,
        triad_code,
        township_name,
        township_code,
        nbhd,
        municipality_name,
        class,
        major_class,
        property_group,
        reassessment_year
    FROM "z_ci_424_identify_ahsap_properties_and_update_reporting_views_reporting".vw_pin_township_class
)

SELECT * FROM old
EXCEPT
SELECT * FROM new
```

And that's what it does.

**Comment by wrridgeway** [comment]
The only thing to test here other than that the changes didn't alter the structure of `vw_pin_township_class` is the `ahsap` column. There really isn't a great way to do that other than to compare against the list of AHSAP parcels on the intranet. I did that and while there's a decent amount of overlap, there are differences. I sent PINs from the intranet not captured by this iasWorld logic to Brian so he can investigate. For now, this is as good as our indicator is going to get and we can update it once Brian gets back to us with any new information.

File: dbt/models/reporting/reporting.vw_pin_township_class.sql [hunk 1]
```diff
@@ -9,6 +9,59 @@ WITH correct_class AS (
         deactivat,
         cur
     FROM {{ source('iasworld', 'pardat') }}
+),
+
+/* This CTE creates our AHSAP indicator. It's a function of multiple columns
```
Comment (dfsnow) [inline dbt/models/reporting/reporting.vw_pin_township_class.sql:14]: **suggestion (non-blocking):** Maybe add a small bit of context here for those unfamiliar with the AHSAP program and/or why the indicator is so complicated.

Comment (dfsnow) [inline dbt/models/reporting/reporting.vw_pin_township_class.sql:14]: suggestion (non-blocking): Maybe add a small bit of context here for those unfamiliar with the AHSAP program and/or why the indicator is so complicated.

File: dbt/models/reporting/reporting.vw_pin_township_class.sql [hunk 2]
```diff
@@ -9,6 +9,59 @@ WITH correct_class AS (
...
+TRUE > FALSE). user columns with "AI" prefixes that trigger AHSAP status are
+'alternative CDU' and user columns with "SAP" prefixes that trigger AHSAP status
+are 'incentive number'. comdat.user13 is 'subclass 2'. */
+ahsap AS (
+    SELECT
+        par.parid,
+        par.taxyr,
+        MAX(COALESCE(
+            land.infl1 IN ('30', '31', '32') OR SUBSTR(land.user7, 1, 3) = 'SAP'
+            OR SUBSTR(oby.user16, 1, 2) = 'AI'
```
Comment (dfsnow) [inline dbt/models/reporting/reporting.vw_pin_township_class.sql:28]: What if we need a new CDU for AI generated buildings..?

**Review by dfsnow — APPROVED** [review]
Disregarding the absurdity of tracking this in 8 different columns across 5 different tables, it's pretty neat that we now have an indicator for this. Nice work.

**Pull request merged**